src/main/java/com/spotinst/sdkjava/model/service/aws/managed/instance/AwsManagedInstanceService.java

```diff
@@ -0,0 +1,337 @@
+
+        Map<String, String> queryParams = new HashMap<>();
+
+        // Add account Id Query param
+        if (account != null) {
+            queryParams.put("accountId", account);
+        }
+
+        // Add managedInstanceId Query param
+        if (managedInstanceId != null) {
+            queryParams.put("MI_ID", managedInstanceId);
+        }
+
+        // Get the headers
+        Map<String, String> headers = buildHeaders(authToken);
+
+        // Write to json
+        Map<String, ApiManagedInstance> managedInstanceUpdateRequest = new HashMap<>();
+        managedInstanceUpdateRequest.put("managedInstance", managedInstanceToUpdate);
+        String body = JsonMapper.toJson(managedInstanceUpdateRequest);
+
+        // Build URI
+        String uri = String.format("%s/aws/ec2/managedInstance/%s", apiEndpoint, managedInstanceId);
+
+        // Send the request
+        RestResponse response = RestClient.sendPut(uri, body, headers, queryParams);
+
+        // Handle the response.
+        AwsManagedInstanceApiResponse managedInstanceUpdateResponse =
+                getCastedResponse(response, AwsManagedInstanceApiResponse.class);
+
+        if (managedInstanceUpdateResponse.getResponse().getStatus().getCode() == HttpStatus.SC_OK) {
+            retVal = true;
+        }
+
+        return retVal;
+    }
+
+    public static ApiManagedInstance getManagedInstance(String managedInstanceId, String authToken,
+                                                              String account) throws SpotinstHttpException {
+        // Init retVal
+        ApiManagedInstance retVal = null;
+
+        // Get endpoint
+        SpotinstHttpConfig config      = SpotinstHttpContext.getInstance().getConfiguration();
+        String             apiEndpoint = config.getEndpoint();
+
+        Map<String, String> queryParams = new HashMap<>();
+
+        // Add account Id Query param
+        if (account != null) {
+            queryParams.put("accountId", account);
+        }
+
+        // Add managedInstanceId Query param
+        if (managedInstanceId != null) {
+            queryParams.put("MI_ID", managedInstanceId);
+        }
+
+        // Get the headers
+        Map<String, String> headers = buildHeaders(authToken);
+
+        // Build URI
+        String uri = String.format("%s/aws/ec2/managedInstance/%s", apiEndpoint, managedInstanceId);
+
+        // Send the request
+        RestResponse response = RestClient.sendGet(uri, headers, queryParams);
+
+        // Handle the response.
+        AwsManagedInstanceApiResponse managedInstanceGetResponse =
+                getCastedResponse(response, AwsManagedInstanceApiResponse.class);
+
+        if (managedInstanceGetResponse.getResponse().getCount() > 0) {
+            retVal = managedInstanceGetResponse.getResponse().getItems().get(0);
+        }
+
+        return retVal;
+    }
+
+    public static List<ApiManagedInstance> getAllManagedInstance(String authToken,
```

Comment:
Rename function to getAllManagedInstances

Reply:
changed